src/openrct2/ride/RideData.h

```diff
@@ -104,33 +103,76 @@ struct RideTypeDescriptor
+{
+    RIDE_TYPE_FLAG_HAS_TRACK_COLOUR_MAIN = 1ull << 0ull,
+    RIDE_TYPE_FLAG_HAS_TRACK_COLOUR_ADDITIONAL = 1ull << 1ull,
+    RIDE_TYPE_FLAG_HAS_TRACK_COLOUR_SUPPORTS = 1ull << 2ull,
+    RIDE_TYPE_FLAG_HAS_SINGLE_PIECE_STATION = 1ull << 3ull, // Set by flat ridesull, tower rides and shops/stalls.
+    RIDE_TYPE_FLAG_HAS_LEAVE_WHEN_ANOTHER_VEHICLE_ARRIVES_AT_STATION = 1ull << 4ull,
+    RIDE_TYPE_FLAG_CAN_SYNCHRONISE_ADJACENT_STATIONS = 1ull << 5ull,
+    RIDE_TYPE_FLAG_TRACK_MUST_BE_ON_WATER = 1ull << 6ull, // used only by boat Hire and submarine ride
+    RIDE_TYPE_FLAG_HAS_G_FORCES = 1ull << 7ull,
+    RIDE_TYPE_FLAG_CANNOT_HAVE_GAPS = 1ull
+        << 8ull, // used by rides that can't have gaps, like those with a vertical tower, such as the observation tower
+    RIDE_TYPE_FLAG_HAS_DATA_LOGGING = 1ull << 9ull,
+    RIDE_TYPE_FLAG_HAS_DROPS = 1ull << 10ull,
+    RIDE_TYPE_FLAG_NO_TEST_MODE = 1ull << 11ull,
+    RIDE_TYPE_FLAG_TRACK_ELEMENTS_HAVE_TWO_VARIETIES = 1ull
+        << 12ull, // used by rides with two varieties, like the u and o shapes of the dinghy slide and the dry and submerged
+                  // track
+    // of the water coaster
+    RIDE_TYPE_FLAG_NO_VEHICLES = 1ull << 13ull, // used only by maze, spiral slide and shops
+    RIDE_TYPE_FLAG_HAS_LOAD_OPTIONS = 1ull << 14ull,
+    RIDE_TYPE_FLAG_HAS_NO_TRACK = 1ull << 15ull,
+    RIDE_TYPE_FLAG_VEHICLE_IS_INTEGRAL = 1ull
+        << 16ull, // Set by flat rides where the vehicle is integral to the structureull, like
+    // Merry-go-round and swinging ships. (Contrast with rides like dodgems.)
+    RIDE_TYPE_FLAG_IS_SHOP = 1ull << 17ull,
+    RIDE_TYPE_FLAG_TRACK_NO_WALLS = 1ull << 18ull, // if setull, wall scenery can not share a tile with the ride's track
+    RIDE_TYPE_FLAG_FLAT_RIDE = 1ull << 19ull,
+    RIDE_TYPE_FLAG_PEEP_WILL_RIDE_AGAIN = 1ull
+        << 20ull, // whether or not guests will go on the ride again if they liked it (this is
+    // usually applied to everything apart from transport rides)
+    RIDE_TYPE_FLAG_PEEP_SHOULD_GO_INSIDE_FACILITY = 1ull
+        << 21ull, // used by toilets and first aid to mark that peep should go inside
+    // the building (rather than 'buying' at the counter)
+    RIDE_TYPE_FLAG_IN_RIDE = 1ull << 22ull, // peeps are "IN" (ride) rather than "ON" (ride)
+    RIDE_TYPE_FLAG_SELLS_FOOD = 1ull << 23ull,
+    RIDE_TYPE_FLAG_SELLS_DRINKS = 1ull << 24ull,
+    RIDE_TYPE_FLAG_IS_TOILET = 1ull << 25ull,
+    RIDE_TYPE_FLAG_HAS_VEHICLE_COLOURS = 1ull << 26ull, // whether or not vehicle colours can be set
+    RIDE_TYPE_FLAG_CHECK_FOR_STALLING = 1ull << 27ull,
+    RIDE_TYPE_FLAG_HAS_TRACK = 1ull << 28ull,
+    RIDE_TYPE_FLAG_ALLOW_EXTRA_TOWER_BASES = 1ull << 29ull, // Only set by lift
+    RIDE_TYPE_FLAG_HAS_LARGE_CURVES = 1ull << 30ull,        // whether the ride supports large (45 degree turn) curves
+    RIDE_TYPE_FLAG_SUPPORTS_MULTIPLE_TRACK_COLOUR = 1ull << 31ull,
+
+    RIDE_TYPE_FLAG_ALLOW_DOORS_ON_TRACK = 1ull << 32ull,
```

Comment:
I'd like to suggest using `ULL` instead of `ull`. In the latter, the 1 and l look awfully similar in many monotype fonts, unfortunately. (This was discussed before, but I can't find it for reference right now.)